lib/rules/no-dataset.js

```diff
@@ -0,0 +1,11 @@
+module.exports = function(context) {
+  return {
+    "Identifier": function(node) {
```

Comment:
Cool. I made that change in https://github.com/github/eslint-plugin-github/pull/13/commits/2722d990d8751ccaeddb9ba388ddc782f256949c and added some additional test cases.